subprojects/testing-jvm-infrastructure/build.gradle.kts

```diff
@@ -41,6 +41,16 @@ dependencies {
 
     testImplementation(testFixtures(project(":core")))
     testImplementation(testFixtures(project(":messaging")))
+    testImplementation(libs.assertj) {
+        because("We test assertion errors coming from AssertJ")
+    }
+    testImplementation("org.opentest4j:opentest4j") {
+        version {
+            // MultipleFailuresError appears only since 1.3.0-RC2
+            require("1.3.0-RC2")
+        }
+        because("We test assertion errors coming from OpenTest4J")
+    }
```

Comment:
❓ My understanding is that the Gradle code only reflectively interacts with opentest4j classes. Why do we need the dependency here?

Reply:
This is `testImplementation`; we instantiate a failure in the tests.
https://github.com/gradle/gradle/blob/1fdeaddce73e2884d783a2e138e1b7897ad75713/subprojects/testing-jvm-infrastructure/src/test/groovy/org/gradle/api/internal/tasks/testing/failure/mappers/OpentestMultipleFailuresMapperTest.groovy#L30-L37

Reply:
Ah, my bad. All clear now.